docs/trace/links-creation-with-new-activities/Program.cs

```diff
@@ -0,0 +1,87 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+namespace LinksCreationWithNewRootActivitiesDemo;
+
+internal class Program
+{
+    private static readonly ActivitySource MyActivitySource = new("LinksCreationWithNewRootActivities");
+
+    public static void Main(string[] args)
+    {
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource("LinksCreationWithNewRootActivities")
+                .AddConsoleExporter()
+                .Build();
+
+        using (var activity = MyActivitySource.StartActivity("OrchestratingActivity"))
+        {
+            activity?.SetTag("foo", 1);
+            DoFanout();
+
+            using (var nestedActivity = MyActivitySource.StartActivity("WrapUp"))
+            {
+                nestedActivity?.SetTag("foo", 1);
+            }
+        }
+    }
+
+    public static void DoFanout()
+    {
+        var previous = Activity.Current;
+        const int NumConcurrentOperations = 10;
+
+        var activityContext = Activity.Current!.Context;
+        var links = new List<ActivityLink>
+        {
+            new ActivityLink(activityContext),
+        };
+
+        // Fanning out to N concurrent operations.
```

Comment:
I don't know `concurrent` is the best term to use here as it one might conflate it with sequential/concurrent nature of execution. Maybe use something like this?
```suggestion
        // Fanning out to N related but individually unique traces.
```

Reply:
While I see the point about why this must be specified as concurrent (as a similar thing could apply for sequential too), I would prefer to leave it in the current form (as it is simpler to understand). My reasoning: The scenario is about a high number of fanout calls that affect individual resources, so realistically this would have to be done concurrently (as doing it sequentially for a high number of calls would make it very slow and further is not needed as each such fanout call is independent). Please feel free to let me know if you disagree.

Reply:
My take would be to just write a comment saying in a real-world case, these "related but individually unique traces" would be spawned concurrently. I believe that using the term "concurrent" could mislead a novice user to refer to the example code as a way of achieving concurrent creation of activities on C#.

If the intent here is to indeed emphasize on concurrent nature of these operations, then maybe we could update the example to use Threads/Tasks to show that instead of using a sequential for loop.

Reply:
Thanks @utpilla , that makes sense. 

I have now updated the example to launch each of these operations concurrently (instead of the sequential fashion). Please let me know if this looks better. 

As of now, I have not changed the comment (but can tweak it if you think it is required) - now it reflects the reality better as it is now triggering these operations concurrently (and these new activities also get created concurrently).